### DISCUSSION TITLE
Gitea directory icon

### DISCUSSION THREAD

**Discussion by schr1k** [jonathanlermitage/intellij-extra-icons-plugin#184] [Ideas]
Icon for .gitea directory (simmilar to .github) and workflows inside workflows subdirectory

**Comment by jonathanlermitage** [comment]
Good idea! I will add it for the next plugin release.

**Reply by schr1k** [reply]
Thanks a lot!!!

**Comment by jonathanlermitage** [comment]
Hi @schr1k  
I just packaged a preliminary version with Gitea support: [Extra Icons-2024.6.3.zip](https://github.com/user-attachments/files/16392178/Extra.Icons-2024.6.3.zip). Can you please tell me if this is OK for you, or if it should be reworked?  
To install this package, go to File > Settings > Plugins > the wheel icon > Install plugin from disk.  

For the old UI, I added two variants for the Gitea folder icon: one gray (default), and one green. You can choose your preferred icon by going to File > Appearance & Behavior > Extra Icons > Plugin Icons > and unselect the gray icon if you want:  
![image](https://github.com/user-attachments/assets/f0c13a5f-240b-4917-a795-afa6d301326b)

For the new UI, both icons are similar. The second icon is just a bit brighter.

For the workflow actions, I registered an icon for yml files in `.gitea/workflows/`. This is the green icon, with a green arrow (like the github actions icon).

**Comment by jonathanlermitage** [comment]
Hum, when using the new UI, maybe the second icon should be white (and black when using a light theme)? The two available icons would be:  
- dark theme:
![image](https://github.com/user-attachments/assets/29943260-736b-4230-9b64-6bd3df8a52ab)

- light theme:
![image](https://github.com/user-attachments/assets/6b7986e5-b028-499d-8dbf-d4130c136f27)

[Extra Icons-2024.6.3.zip](https://github.com/user-attachments/files/16392264/Extra.Icons-2024.6.3.zip)

**Comment by schr1k** [comment]
Hey @jonathanlermitage
I think the main variant of .gitea directory icon is too dark (second screenshot), and alternative is ok (first screenshot).
Maybe u should set brighter green one as main and white one as alternative.
Also is it possible to detect both *.yml and *.yaml files inside workflows directory?
Overall looks incredible!
![image](https://github.com/user-attachments/assets/fd4de49e-edc2-49eb-80d8-c1890f7e240f)
![image](https://github.com/user-attachments/assets/ff2c657b-feab-4dbf-a288-63c663d7188e)

**Comment by jonathanlermitage** [comment]
Sure, here is a new build with the new folder icon with a brighter green logo, and support for YAML files: 
[Extra Icons-2024.6.3.zip](https://github.com/user-attachments/files/16392972/Extra.Icons-2024.6.3.zip)

Thank you very much for your feedback!

**Reply by schr1k** [reply]
Wonderful, i think u should add yaml support for github and gitlab too

**Comment by jonathanlermitage** [comment]
I would have a question. I did not use GitLab since a while, and I can't find if the `.gitlab` directory can define tasks (with yaml files) in a `workflows` subdirectory. Do you confirm I should show a custom icon for y(a)ml files in `.gitlab/workflows/`? Or every yaml file inside `.gitlab` and its subdirectories? I only remember I was using a `.gitlab-ci.yml` file.
Thanks

**Reply by schr1k** [reply]
I've never used gitlab, i just thought they use simmilar mechanism for ci/cd.
After research i see they use `.gitlab-{name}.yml` files as you can see [here](https://docs.gitlab.com/ee/ci/) but i think yaml should do the thing too.

**Reply by jonathanlermitage** [reply]
No problem. Thx for the link!  
Now I remind how it works. The tasks are registered in the `.gitlab-ci.yml`, and we can give them any name. I will see later if I can parse this file.
